test/test.js

```diff
@@ -0,0 +1,97 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+let accounts;
+let testToken;
+let minter;
+
+describe("Minter", async function () {
+
+  async function init(){
+    accounts = await ethers.getSigners();
+
+    const TestToken = await hre.ethers.getContractFactory("TestToken");
+    testToken = await TestToken.deploy();
+    await testToken.deployed();
+
+    let currentBlockNumber = await ethers.provider.getBlockNumber()
+    const Minter = await hre.ethers.getContractFactory("Minter");
+    minter = await Minter.deploy(testToken.address, currentBlockNumber + 1, accounts[0].address);
+    await minter.deployed();
+  }
+
+  it("should add pool", async function () {
+    await init()
+
+    let add = await minter.addPool(accounts[0].address, 1);
+    await add.wait();
+
+    let getPool = await minter.getPool(0);
+    let getPoolLength = await minter.getPoolLength();
+
+    expect(getPool.receiver).to.equal(accounts[0].address)
+    expect(getPool.amountPerBlock.toNumber()).to.equal(1)
+    expect(getPoolLength.toNumber()).to.equal(1)
+  });
+
+  it("should mint tokens", async function () {
```

Comment:
Recommend expanding the test cases here significantly. Should include edge cases (0 pools, 1 pools, many pools, 0 blocks elapsed, etc...). Should verify minting works before and after updates, removes, and adds. Should test the edge cases around reaching cap.

Reply:
More tests added: https://github.com/steem-monsters/sps-minter/commit/4122b13ddb24863f8bc789af0fb316604d94be4f, https://github.com/steem-monsters/sps-minter/commit/fdd8e7977859e472082d4f7221b177f1db794f99

Reply:
New tests look good. Since the cap logic is sensitive, I would recommend adding one explicitly that:

* Has multiple minters
* Reaches the cap while minting
* Gives some tokens to some minters
* Gives 0 tokens to other minters
* Reverts immediately on the next call to mint.

Reply:
- Has multiple pools: https://github.com/steem-monsters/sps-minter/commit/94d92773c393d236ea0cc8ad8ef0a3e4b85a2284
- Reaches cap while minting: should be already covered here: https://github.com/steem-monsters/sps-minter/blob/re-add/test/test.js#L99
- Gives some tokens to some minters: should be already covered here: https://github.com/steem-monsters/sps-minter/blob/re-add/test/test.js#L169
- Gives 0 tokens to other minters: https://github.com/steem-monsters/sps-minter/commit/d7b5e0024c38d3efd9f1d0161c68778dd7fc92e8
- Reverts immediately on the next call to mint: should be already covered here: https://github.com/steem-monsters/sps-minter/blob/re-add/test/test.js#L118